File: Array/two-sum.cpp
/* Question: https://leetcode.com/problems/two-sum/

Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and you may not use the same element twice.

You can return the answer in any order.

Solution */ 

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int,int> map;
        for(int i=0; i<nums.size();i++){ 
            if(map.find(target-nums[i]) != map.end()){
                res.push_back(map[target-nums[i]]);
                res.push_back(i);
            }
            map[nums[i]] = i;    
        }
        return res;
    }
};

File: Array/two-sum.py
'''
https://leetcode.com/problems/two-sum/
Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
You may assume that each input would have exactly one solution, and you may not use the same element twice.
You can return the answer in any order.

Solution :
'''


class Solution(object):
  
	def twoSum(self, nums, target):
    # creating a dict
		map_elements = {}
		for i in range(len(nums)):
      # finding elements in the dict if already present
			if nums[i] in map_elements:
				return [map_elements[nums[i]], i] 
			else:
        # add required number to pair
				map_elements[target - nums[i]] = i         
